From e21cf38380aaab00e3e8c5601c4d148314958ab7 Mon Sep 17 00:00:00 2001
From: fhoehle <fhoehle@users.noreply.github.com>
Date: Sat, 14 Jun 2025 08:00:09 +0200
Subject: [PATCH] fix type annotation to be 3.9 compliant

---
 signxml/verifier.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/signxml/verifier.py b/signxml/verifier.py
index 4ca6caa..29e3fa8 100644
--- a/signxml/verifier.py
+++ b/signxml/verifier.py
@@ -133,7 +133,7 @@ class XMLVerifier(XMLSignatureProcessor):
         key_value: Optional[etree._Element] = None,
         der_encoded_key_value: Optional[etree._Element] = None,
         signing_certificate: Optional[x509.Certificate] = None,
-    ) -> Tuple[bytes, bytes | dsa.DSAPublicKey | rsa.RSAPublicKey | ec.EllipticCurvePublicKey]:
+    ) -> Tuple[bytes, Union[bytes, dsa.DSAPublicKey, rsa.RSAPublicKey, ec.EllipticCurvePublicKey]]:
         if der_encoded_key_value is not None:
             assert der_encoded_key_value.text is not None
             key = load_der_public_key(b64decode(der_encoded_key_value.text))
-- 
2.49.0

